File: micronaut-mcp-server-java-sdk/src/test/java/io/micronaut/mcp/server/tools/search/MicronautModulesSearch.java
package io.micronaut.mcp.server.tools.search;

import io.micronaut.context.annotation.Requires;
import io.micronaut.mcp.server.context.MicronautMcpTransportContext;
import jakarta.inject.Singleton;

import java.util.List;

@Requires(property = "spec.tool.name", value = "MicronautModulesSearch")
//tag::clazz[]
@Singleton
class MicronautModulesSearch implements SearchTool {

    @Override
    public SearchResponse search(SearchRequest request, MicronautMcpTransportContext transportContext) {
        return new SearchResponse(List.of(SearchResult.builder()
            .id("micronaut-security")
            .title("Micronaut Security")
            .url("https://micronaut-projects.github.io/micronaut-security/latest/guide")
            .build()));
    }
}
//end::clazz[]

File: test-suite-docs-kotlin/src/test/kotlin/io/micronaut/mcp/docs/tools/search/MicronautModulesSearch.kt
package io.micronaut.mcp.docs.tools.search

//tag::imports[]
import io.micronaut.context.annotation.Requires
import io.micronaut.mcp.server.tools.search.SearchRequest
import io.micronaut.mcp.server.tools.search.SearchResponse
import io.micronaut.mcp.server.tools.search.SearchResult
import io.micronaut.mcp.server.tools.search.SearchTool
import io.modelcontextprotocol.common.McpTransportContext
import jakarta.inject.Singleton

//end::imports[]
@Requires(property = "spec.name", value = "MicronautModulesSearchTest")
//tag::clazz[]
@Singleton
class MicronautModulesSearch : SearchTool {

    override fun search(request: SearchRequest, transportContext: McpTransportContext?): SearchResponse {
        return SearchResponse(listOf(SearchResult.builder()
            .id("micronaut-security")
            .title("Micronaut Security")
            .url("https://micronaut-projects.github.io/micronaut-security/latest/guide")
            .build()))
    }
}
//end::clazz[]
